src/nerdbank-streams/src/Channel.ts

```diff
@@ -0,0 +1,135 @@
+import { Duplex } from "stream";
+import { ChannelOptions } from "./ChannelOptions";
+import { ControlCode } from "./ControlCode";
+import { Deferred } from "./Deferred";
+import { FrameHeader } from "./FrameHeader";
+import { IDisposableObservable } from "./IDisposableObservable";
+import { MultiplexingStreamClass } from "./MultiplexingStream";
+
+export abstract class Channel implements IDisposableObservable {
+    public duplex: Duplex;
+    public acceptance: Promise<void>;
+    public completion: Promise<void>;
+    private _isDisposed: boolean = false;
+
+    get isDisposed(): boolean {
+        return this._isDisposed;
+    }
+
+    public dispose() {
+        this._isDisposed = true;
+    }
+}
+
+// tslint:disable-next-line:max-classes-per-file
+export class ChannelClass extends Channel {
+    public readonly id: number;
+    private _duplex: Duplex;
```

Comment:
Why not call this stream, or _duplexStream if you're unhappy with the name?

Reply:
Renamed it to `stream`.

Reply:
I also changed its exposed type to be `NodeJS.ReadWriteStream`, since `Duplex` is (oddly) in an `internal` namespace.